packages/react-instantsearch-hooks-dom/src/ui/PoweredBy.tsx

```diff
@@ -1,21 +1,57 @@
 import React from 'react';
-import { usePoweredBy } from 'react-instantsearch-hooks';
 
-export function PoweredBy() {
-  const { url } = usePoweredBy();
+import { cx } from './lib/cx';
 
+export type PoweredByClassNames = {
+  /**
+   * Class names to apply to the root element
+   */
+  root: string;
+  /**
+   * Class names to apply to the link element
+   */
+  link: string;
+  /**
+   * Class names to apply to the SVG logo
+   */
+  logo: string;
+};
+
+export type PoweredByProps = React.HTMLAttributes<HTMLDivElement> & {
+  classNames: Partial<PoweredByClassNames>;
+  url: string;
+  theme?: 'light' | 'dark';
+};
+
+export function PoweredBy({
+  classNames,
+  url,
+  theme = 'light',
+  ...props
+}: PoweredByProps) {
   return (
-    <div className="ais-PoweredBy">
+    <div
+      {...props}
+      className={cx(
+        'ais-PoweredBy',
+        theme === 'dark' ? 'ais-PoweredBy--dark' : 'ais-PoweredBy--light',
```

Comment:
```suggestion
        theme === 'dark' ? 'ais-PoweredBy--dark' : 'ais-PoweredBy--light',
        theme === 'dark' ? classNames.dark : classNames.light,
```

Reply:
There's an inconsistency between [InstantSearch.js](https://www.algolia.com/doc/api-reference/widgets/powered-by/js/?client=vue#widget-param-cssclasses) and [Vue InstantSearch](https://www.algolia.com/doc/api-reference/widgets/powered-by/vue/?client=react#widget-param-class-names) regarding those CSS classes. **What should we follow as the source of truth?**

The benefit of following Vue InstantSearch is that we perform the condition for the user, that's less code for them. If we agree on this, I'll also make the change in InstantSearch.js to avoid further drifts between flavors.

Reply:
```suggestion
        theme === 'dark' ? cx(classNames.dark, 'ais-PoweredBy--dark') : cx(classNames.light, 'ais-PoweredBy--light'),
```

alternative option to that condition.



IMO we should make sure every class name is modifiable, which is useful for eg. tailwind

Reply:
Yes I much prefer this option rather than doing the condition twice. Addressed in 2f8451e3.